chalice/deploy/newdeployer.py

```diff
@@ -275,8 +297,7 @@ def _traverse(self, resource, ordered, seen):
             if id(dep) not in seen:
                 seen.add(id(dep))
                 self._traverse(dep, ordered, seen)
-        if resource not in ordered:
-            ordered.append(resource)
+        ordered.append(resource)
```

Comment:
At first glance I want to say it's not possible because `id(resource)` is checked before you make the recursive call, thus you'll never call traverse on the same resource twice.

Reply:
Actually, it is possible if there's a dependency on the first node in one of its children. `A -> B -> A`, for example. Though, I'm not sure this would ever happen? Simply adding the first node to the seen set before you call traverse would fix it though.

Reply:
It is possible. It gets broken with the case where say there are two resources in the top level resource `graph`: A and B. If A has a dependency on B, it will add B to the list making the list [B, A]. But then B is blindly added to the list as the `if id(dep) not in seen` check is done in the for loop, which would make the list [B, A, B]. This can be fixed though if graph is directly passed into `_traverse()`, but that means the graph would be included in the ordered list. Or the original if statement can be added back.

Reply:
Capturing offline discussion.  I'll add the old statement back.  It needs to change slightly because we're checking with identity instead of equality so I have to check that the resource's id isn't an id that's in `[id(r) for r in resource]`.